### ISSUE TITLE
Requires cycle warning

### ISSUE THREAD

**Issue by shirshak55** [zalmoxisus/mobx-remotedev#36]
It uses jsan packages which shows warning like this.

```
require.js:122 Require cycle: node_modules/jsan/lib/index.js -> node_modules/jsan/lib/cycle.js -> node_modules/jsan/lib/utils.js -> node_modules/jsan/lib/index.js
```

**Comment by zalmoxisus** [comment]
Related to https://github.com/kolodny/jsan/issues/25. Looks like that limitation was added somewhere in React Native, it doesn't have any issues, but better to handle on `jsan` part.

**Comment by zalmoxisus** [comment]
This is from React Native, right? If I'd have a repro would look into it, but if someone can do a PR on `jsan` part would be much appreciated. Or waiting till Metro decides to make it opt in as per https://github.com/facebook/metro/issues/287.

**Comment by shirshak55** [comment]
@zalmoxisus when i open try to debug react native using remote devtool it shows unknown call relay:check .and it shows following thing on devtools. 

In screenshot you can see the message too.

<img width="1792" alt="screen shot 2018-12-02 at 12 51 45 pm" src="https://user-images.githubusercontent.com/8097377/49336800-115df200-f631-11e8-92bb-501cbc812e5e.png">


Thanks for concern :)


I have added you in repo so you can investigate.

Thanks again.

**Comment by shirshak55** [comment]
@zalmoxisus when i open try to debug react native using remote devtool it shows unknown call relay:check .and it shows following thing on devtools. 

In screenshot you can see the message too.

<img width="1792" alt="screen shot 2018-12-02 at 12 55 06 pm" src="https://user-images.githubusercontent.com/8097377/49336838-a103a080-f631-11e8-8207-400d26db0760.png">



I have added you in repo so you can investigate.

Thanks again.

**Comment by zalmoxisus** [comment]
Thanks for adding me. Unfortunately I'm not using RN atm and don't have an emulator on this machine to check now. Could you try to replace `./node-modules/jasn/lib/utils.js` and `restore-special.js` inside your project form [here](https://github.com/kolodny/jsan/tree/require-cycle/lib). Most likely not.

I see it's warning not only when requiring the lib so should be quite annoying, but it would need to restructure the whole lib into one file which isn't quite sustainable just for a linting warning as discussed in https://github.com/kolodny/jsan/issues/25.

**Comment by shirshak55** [comment]
@zalmoxisus i replaced it but it seems everything is same. I needed to create new file for `restore-special.js` file.

**Comment by zalmoxisus** [comment]
Yes, I tried to move it to new `restore-special.js` file, thought it would ignore the cycle in run time this way, but seems no.

**Comment by shirshak55** [comment]
@zalmoxisus is it due to package issue or which one? like many package etc are starting to showing up. If i don't use devtool it doesn't show any errors.

**Comment by zalmoxisus** [comment]
Yes, we have that in `jsan`, see https://github.com/kolodny/jsan/issues/25 for more details. But that's not quite an issue with the package itself, it works as expected, but Metro from React Native is warning on that. As many suggested in https://github.com/facebook/metro/issues/287, that should be be opt-in.
